CMSgov/mint-app
# [EASI-2764] Search Change History Proof of Concept

# EASI-2764

## Changes and Description

- Added Logstash, OpenSearch, OpenSearch-Dashboards
- Added Logstash data ingestion
- Added endpoints to search on OpenSearch
  - SearchChangeTable
    - Accepts an Elasticsearch DSL query
  - SearchChangeTableWithFreeText
  - SearchChangeTableByModelPlanID
  - SearchChangeTableByDateRange
  - SearchChangeTableByActor
  - SearchChangeTableByModelStatus



## How to test this change

### Initialize Your Environment
- Run [scripts/set_vm_map.sh](https://github.com/CMSgov/mint-app/pull/518/files#diff-171c3036c8f86feab777a6cc9b100bde502a47facf5a7f1503f9dc41de068df7)

### Test Endpoints
- Open Postman
- Open MINT / Search filter
- Run each endpoint and confirm they function as expected

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
